### PR TITLE
Archive -> Buffer

### PR DESCRIPTION
**Pull request by michael**

### CODE REVIEW

**Comment by oliver----** [comment]
@michael Didn't you realize that you have broken the TestBackend here?
See how, the test coverage dropped.

**Comment by oliver----** [comment]
I suggest, when the test run indicates such an unexpected effect, pls wait with merging and rather ask for help if you don't understand the reason.

**Comment by oliver----** [comment]
Pls fix it.

**Comment by michael** [comment]
Noticed the decrease, but couldn't explain what caused it, tests looked ok. We looked at it together and decided to merge, since we wanted to push a preview release to test the desktop integration.

You have a pointer how to fix?

**Comment by oliver----** [comment]
You have refactored TestBackend. So go through the code and see where you have broken it.

File: tests/backend/TestBackend.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-forEach(testVFS, (content, address) => {
-  if (/\.html$/.exec(address)) {
-    testLibrary[address] = {
+forEach(testVFS, (content, documentId) => {
+  // Only pick HTML documents
+  if (/\.html$/.exec(documentId)) {
+    testLibrary[documentId] = {
       type: 'document',
-      address,
-      title: address,
```
Comment (oliver----) [inline tests/backend/TestBackend.js]: here you rename a property. Have you renamed it throughout the whole file?

File: tests/backend/TestBackend.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ export default class TestBackend extends MemoryBackend {
     })
   }
 
-  _getDocumentUrls() {
+  _getDocumentIds() {
```
Comment (oliver----) [inline tests/backend/TestBackend.js]: This is used by tests to traverse all documents. You probably have broken it.

**Comment by oliver----** [comment]
You might want to add a test just for checking the validity of the TestBackend.

**Comment by oliver----** [comment]
> Noticed the decrease, but couldn't explain what caused it, tests looked ok. We looked at it together and decided to merge, since we wanted to push a preview release to test the desktop integration.

While I understand the desire to move on quickly, I don't like that you decided this silently. It did cost me some time to figure out.

**Comment by michael** [comment]
Fixed via #245

**Comment by michael** [comment]
Will add a note on such things next time. Sorry.

**Comment by oliver----** [comment]
Thx 👍

**Pull request merged**